[Bug]: Debian/Ubuntu repo postrm/postinst scripts might need some love

For your issue to not get closed without review, please confirm that:

- [ x] This issue does not fit into any of the predefined categories, which is
      why I am making a blank issue from scratch.
- [x ] I am not reporting a security vulnerability through this issue, because
      I am aware that there is an appropriate channel for that.
- [x] I understand that I will be permanently banned from interacting with this
      organization if I lied by checking any of these checkboxes.

---

[your issue text goes here]
Disclaimer: I couldn't find a better/more suitable place to report this.

When using the Debian/Ubuntu repo to install/upgrade helium-bin, I noticed apt complaining about /var/lib/dpkg/info/helium-bin.postrm:

```console
$ sudo apt install helium-bin
[...]
Unpacking helium-bin (0.11.3.2-1) over (0.10.9.1-1) ...
/var/lib/dpkg/info/helium-bin.postrm: 4: [: Illegal number: upgrade
[...]
```

So I turned to shellcheck to see what that had to report on that file (and on /var/lib/dpkg/info/helium-bin.postinst too for good measure). Looks like there's some non-Posix stuff in there that might be worthwhile looking into.

<details>
<summary> $ shellcheck /var/lib/dpkg/info/helium-bin.postrm </summary>

    ```
    In /var/lib/dpkg/info/helium-bin.postrm line 3:
/usr/bin/update-desktop-database &> /dev/null || :
                                                                    ^----------^ SC3020 (warning): In POSIX sh, &> is undefined.


In /var/lib/dpkg/info/helium-bin.postrm line 4:
if [ $1 -eq 0 ] ; then
      ^-- SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean:
if [ "$1" -eq 0 ] ; then


In /var/lib/dpkg/info/helium-bin.postrm line 5:
    /bin/touch --no-create /usr/share/icons/hicolor &>/dev/null
                                                                                               ^---------^ SC3020 (warning): In POSIX sh, &> is undefined.


In /var/lib/dpkg/info/helium-bin.postrm line 6:
    /usr/bin/gtk-update-icon-cache /usr/share/icons/hicolor &>/dev/null || :
                                                                                                                 ^---------^ SC3020 (warning): In POSIX sh, &> is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3020 -- In POSIX sh, &> is undefined.
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
    ```

</details>

<details>
<summary> $ shellcheck /var/lib/dpkg/info/helium-bin.postinst </summary>

    ```
    In /var/lib/dpkg/info/helium-bin.postinst line 13:
/usr/bin/update-desktop-database &> /dev/null || :
                                                                    ^----------^ SC3020 (warning): In POSIX sh, &> is undefined.


In /var/lib/dpkg/info/helium-bin.postinst line 14:
/bin/touch --no-create /usr/share/icons/hicolor &>/dev/null || :
                                                                                           ^---------^ SC3020 (warning): In POSIX sh, &> is undefined.


In /var/lib/dpkg/info/helium-bin.postinst line 21:
/usr/bin/gtk-update-icon-cache /usr/share/icons/hicolor &>/dev/null || :
                                                                                                             ^---------^ SC3020 (warning): In POSIX sh, &> is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3020 -- In POSIX sh, &> is undefined.
    ```

</details>

Additionally, IMO, the apparmor profile handling could be improved. The postinst script runs `apparmor_parser -r`.  The [debian wiki](https://wiki.debian.org/AppArmor/Contribute/FirstTimeProfileImport) recommends also using the '-W -T' options.

Regards

--- a/package/helium-bin.spec
+++ b/package/helium-bin.spec
@@ -88,7 +88,7 @@ ln -sf %{helium_base}/helium-wrapper \
 /usr/bin/update-desktop-database > /dev/null 2>&1 || :
 /bin/touch --no-create %{_datadir}/icons/hicolor > /dev/null 2>&1 || :
 
-if [ -d /etc/apparmor.d ]; then
+if command -v apparmor_parser > /dev/null 2>&1 && [ -d /etc/apparmor.d ]; then
     cp %{helium_base}/apparmor.cfg /etc/apparmor.d/helium-bin
     apparmor_parser -r -W -T /etc/apparmor.d/helium-bin || :
 fi
@@ -102,7 +102,9 @@ case "$1" in
         /usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor > /dev/null 2>&1 || :
 
         if [ -f /etc/apparmor.d/helium-bin ]; then
-            apparmor_parser -R helium-bin || :
+            if command -v apparmor_parser > /dev/null 2>&1; then
+                apparmor_parser -R helium-bin || :
+            fi
             rm -f /etc/apparmor.d/helium-bin
         fi
         ;;

--- a/package/helium-bin.spec
+++ b/package/helium-bin.spec
@@ -103,7 +103,7 @@ case "$1" in
 
         if [ -f /etc/apparmor.d/helium-bin ]; then
             if command -v apparmor_parser > /dev/null 2>&1; then
-                apparmor_parser -R helium-bin || :
+                apparmor_parser -R /etc/apparmor.d/helium-bin || :
             fi
             rm -f /etc/apparmor.d/helium-bin
         fi

--- a/package/helium-bin.spec
+++ b/package/helium-bin.spec
@@ -85,29 +85,31 @@ ln -sf %{helium_base}/helium-wrapper \
 
 %post
 # Refresh icon cache and update desktop database
-/usr/bin/update-desktop-database &> /dev/null || :
-/bin/touch --no-create %{_datadir}/icons/hicolor &>/dev/null || :
+/usr/bin/update-desktop-database > /dev/null 2>&1 || :
+/bin/touch --no-create %{_datadir}/icons/hicolor > /dev/null 2>&1 || :
 
 if [ -d /etc/apparmor.d ]; then
     cp %{helium_base}/apparmor.cfg /etc/apparmor.d/helium-bin
-    apparmor_parser -r /etc/apparmor.d/helium-bin || :
+    apparmor_parser -r -W -T /etc/apparmor.d/helium-bin || :
 fi
 
 %postun
 # Refresh icon cache and update desktop database
-/usr/bin/update-desktop-database &> /dev/null || :
-if [ $1 -eq 0 ] ; then
-    /bin/touch --no-create %{_datadir}/icons/hicolor &>/dev/null
-    /usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
+/usr/bin/update-desktop-database > /dev/null 2>&1 || :
+case "$1" in
+    0|remove|purge)
+        /bin/touch --no-create %{_datadir}/icons/hicolor > /dev/null 2>&1
+        /usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor > /dev/null 2>&1 || :
 
-    if [ -f /etc/apparmor.d/helium-bin ]; then
-        apparmor_parser -R helium-bin || :
-        rm -f /etc/apparmor.d/helium-bin
-    fi
-fi
+        if [ -f /etc/apparmor.d/helium-bin ]; then
+            apparmor_parser -R helium-bin || :
+            rm -f /etc/apparmor.d/helium-bin
+        fi
+        ;;
+esac
 
 %posttrans
-/usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
+/usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor > /dev/null 2>&1 || :
 
 %changelog
 %if "%{_vendor}" != "debbuild"

--- a/scripts/dev.sh
+++ b/scripts/dev.sh
@@ -174,22 +174,22 @@ __helium_menu() {
         reset) ___helium_reset;;
         translate) helium_apply_translations;;
         *)
-            echo "usage: he (setup | build | run | sub | unsub | namesub | nameunsub | merge | unmerge | push | pop | pull | reset | translate)" >&2
-            echo "\tsetup - sets up the dev environment for the first itme" >&2
-            echo "\tbuild - prepares a development build binary" >&2
-            echo "\trun - runs a development build of helium with dev data dir & ui devtools enabled" >&2
-            echo "\tsub - apply google domain and name substitutions" >&2
-            echo "\tunsub - undo google domain substitutions" >&2
-            echo "\tnamesub - apply only name substitutions" >&2
-            echo "\tnameunsub - undo name substitutions" >&2
-            echo "\tmerge - merges all patches" >&2
-            echo "\tunmerge - unmerges all patches" >&2
-            echo "\tpush - applies all patches" >&2
-            echo "\tpop - undoes all patches" >&2
-            echo "\tresources - copies helium resources (such as icons)" >&2
-            echo "\tpull - undoes all patches, pulls, redoes all patches" >&2
-            echo "\treset - nukes everything" >&2
-            echo "\ttranslate - apply translations from i18n folder" >&2
+            printf 'usage: he (setup | build | run | sub | unsub | namesub | nameunsub | merge | unmerge | push | pop | pull | reset | translate)\n' >&2
+            printf '\tsetup - sets up the dev environment for the first time\n' >&2
+            printf '\tbuild - prepares a development build binary\n' >&2
+            printf '\trun - runs a development build of helium with dev data dir & ui devtools enabled\n' >&2
+            printf '\tsub - apply google domain and name substitutions\n' >&2
+            printf '\tunsub - undo google domain substitutions\n' >&2
+            printf '\tnamesub - apply only name substitutions\n' >&2
+            printf '\tnameunsub - undo name substitutions\n' >&2
+            printf '\tmerge - merges all patches\n' >&2
+            printf '\tunmerge - unmerges all patches\n' >&2
+            printf '\tpush - applies all patches\n' >&2
+            printf '\tpop - undoes all patches\n' >&2
+            printf '\tresources - copies helium resources (such as icons)\n' >&2
+            printf '\tpull - undoes all patches, pulls, redoes all patches\n' >&2
+            printf '\treset - nukes everything\n' >&2
+            printf '\ttranslate - apply translations from i18n folder\n' >&2
     esac
 }
 

--- a/scripts/dev.sh
+++ b/scripts/dev.sh
@@ -1,10 +1,16 @@
 # shellcheck disable=SC2148
 
-if [[ "$(basename -- "$0")" = *bash ]]; then
-    . "$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)/shared.sh"
+if [ -n "${BASH_VERSION:-}" ]; then
+    _dev_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+elif [ -n "${ZSH_VERSION:-}" ]; then
+    _dev_dir="${0:a:h}"
 else
-    . "${0:a:h}/shared.sh"
+    echo "dev.sh only supports bash and zsh" >&2
+    return 1 2>/dev/null || exit 1
 fi
+
+. "${_dev_dir}/shared.sh"
+unset _dev_dir
 
 setup_environment
 

--- a/scripts/shared.sh
+++ b/scripts/shared.sh
@@ -2,11 +2,18 @@
 
 # shared build functions used by local and CI scripts
 
+if [ -n "${BASH_VERSION:-}" ]; then
+    __helium_shared_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+elif [ -n "${ZSH_VERSION:-}" ]; then
+    __helium_shared_dir="${0:a:h}"
+else
+    echo "shared.sh only supports bash and zsh" >&2
+    return 1 2>/dev/null || exit 1
+fi
+
 # resolve repo root directory regardless of caller location
 repo_root() {
-    local _base_dir
-    _base_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
-    cd "${_base_dir}/.." >/dev/null 2>&1 && pwd
+    cd "${__helium_shared_dir}/.." >/dev/null 2>&1 && pwd
 }
 
 setup_arch() {

